ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/FunctionInfo.java

```diff
@@ -16,32 +16,39 @@
 
 package io.confluent.ksql.rest.entity;
 
+import com.google.common.collect.ImmutableList;
+
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
 
 public class FunctionInfo {
 
-  private final List<String> argumentTypes;
+  private final List<ArgumentInfo> arguments;
```

Comment:
Good catch. Cc @confluentinc/c3

Reply:
Good catch indeed!

So...  adding a new field to an existing object _should not be a API breaking change_.  I was assuming this `FunctionInfo` class is internal to KSQL. Is this not the case?

On the KSQL side, this is a breaking change, purely because our JSON POJOs are not attributed with `@JsonIgnoreProperties(ignoreUnknown = true)`, which means they throw exceptions when deserialising on unknown properties. This sucks, so I'll fix this in a separate PR.

On the C3 side, it depends how C3 are deserialising the JSON. @confluentinc/c3 are you ignoring unknown properties? To aid forward and backwards compatibility you certainly should be. Do you use our POJOs or your own, or use some other method to parse the JSON?

Then, of course, even without any compatibility C3 should look to support this richer set of information going forward.

Reply:
So my comment was based on me forgetting how jackson populates the json fields. Agree that adding a field is not a breaking change. This also shouldn't break KSQL. We updated the client to not fail on unknown properties as part of the api work for 5.0: https://github.com/confluentinc/ksql/blob/master/ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/KsqlRestClient.java#L365

Reply:
Awesome!

Reply:
This is breaking change in that the new CLI talking to the old server will not work, right? So basically the 5.1 CLI must have a 5.1 servers.

Reply:
Or I guess it forward compatibility is taken care of by #1627 ?